src/Java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMTE_NuclearReactor.java

```diff
@@ -544,87 +524,139 @@ public void explodeMultiblock() {
-						Materials.Fluorine.getGas(fluorineContent)
-				};
+		}
+		if (aSpargeRecipe == null) {
+			return aOutputGases;
+		}		
+		int aSpargeGasAmount = aSpargeRecipe.mInputGas.amount;		
+		FluidStack depletionStack = aSpargeRecipe.mInputGas.copy();
+		depletionStack.amount = aSpargeGasAmount;
+		this.depleteInput(depletionStack);
+		updateSlots();		
+		for (int i=0;i<aSpargeRecipe.mFluidOutputs.length;i++) {
+			if (aSpargeRecipe.mInputGas.isFluidEqual(spargeGas)) {
+				continue; // Skip sparge gas
 			}
+			int aGasAmount = MathUtils.randInt(0, aSpargeRecipe.mMaxOutputQuantity[i]);
+			FluidStack aOutput = aSpargeRecipe.mFluidOutputs[i];
+			aSpargeGasAmount -= aGasAmount;
+			aOutputGases.add(new FluidStack(aOutput.getFluid(), aGasAmount));			
+		}
+		if (aSpargeGasAmount > 0) {
+			aOutputGases.add(new FluidStack(aSpargeRecipe.mInputGas.getFluid(), aSpargeGasAmount));	
 		}
-		return outputArrayOfGases;
+		return aOutputGases;
 	}
 
 	@Override
 	public void onPostTick(IGregTechTileEntity aBaseMetaTileEntity, long aTick) {
-		//Add Power if active
-		if (aBaseMetaTileEntity.isActive()){
-			//this.getBaseMetaTileEntity().increaseStoredEnergyUnits(this.mEUt, false);
-			
-			if (this.mEfficiency >= 500){
-				this.boostEu = true;
-				this.turnCasingActive(true);
+		if (aBaseMetaTileEntity.getWorld().isRemote) {
+			if (aBaseMetaTileEntity.isActive()){
+				// Set casings active if we're warmed up.
+				if (this.mEfficiency == this.getMaxEfficiency(null)){
+					this.turnCasingActive(true);
+				}
+				else {
+					this.turnCasingActive(false);
+				}
 			}
 			else {
-				this.boostEu = false;
 				this.turnCasingActive(false);
 			}
-			
-			if (MathUtils.randInt(1, 200) == 1){
-				//Utils.LOG_INFO("Adding U233");
-				this.addOutput(ELEMENT.getInstance().URANIUM233.getFluid(MathUtils.randInt(1, 10)));
-			}
-			
-			if (this.mDynamoHatches != null) {
-				for (GT_MetaTileEntity_Hatch_Dynamo tHatch : this.mDynamoHatches) {
-					if (tHatch.mTier >= 5){
-						if (isValidMetaTileEntity(tHatch)){
-							tHatch.getBaseMetaTileEntity().increaseStoredEnergyUnits(this.mEUt, false);
-							//Utils.LOG_WARNING("Adding "+this.mEUt+"eu to internal storage of dynamo "+hatchNo+".");
-						}						
-					}
-				}
-			}
-			
 		}
 		else {
-			this.turnCasingActive(false);
-		}
+			// Try output some Uranium-233
+			if (MathUtils.randInt(1, 300) == 1){
+				this.addOutput(ELEMENT.getInstance().URANIUM233.getFluidStack(MathUtils.randInt(1, 10)));
+			}			
+			// Set a random tick counter, count it up.
+			if (this.mSpargeTime == 0) {
+				this.mSpargeTime = MathUtils.randInt(sMinSpargeWait, sMaxSpargeWait);
+				Logger.WARNING("Set Sparge Timer to "+this.mSpargeTime);
+			}
+			else {
+				this.mSpargeTicks++;
+			}
+			// Try Sparge
+			if (this.mSpargeTicks >= this.mSpargeTime) {
+				this.mSpargeTime = 0;
+				this.mSpargeTicks = 0;
+				//Logger.WARNING("Sparging!");
+				//trySparge();
```

Comment:
I suppose you need to uncomment this?

Reply:
Nope, it was originally done in the multi prior to implementing the entire reprocessing chain.

Reply:
Fixed in next push.